content/guides/spring-integration-lock/index.md

```diff
@@ -0,0 +1,366 @@
+---
+date: '2021-10-22'
+lastmod: '2021-10-22'
+title: Working With the Spring Distributed Lock
+linkTitle: Working With the Spring Distributed Lock
+description: A simple demonstration of how to implement a Spring Distributed Lock with in your Spring application
+patterns:
+  - API
+tags:
+  - Spring Boot
+  - Spring Integration
+  - API
+  - Getting Started
+  - Spring
+  - Microservices
+team:
+  - Eric Standley
+weight: 2
+oldPath: '/content/guides/spring/spring-distributed-lock.md'
+aliases:
+  - '/guides/spring/spring-distributed-lock'
+level1: Building Modern Applications
+level2: Frameworks and Languages
+---
+
+When designing microservices, you want to create code to handle multiple instances of services to run at the same time. When you only want to process something one time, or do something similar to the [Leader Pattern](https://docs.microsoft.com/en-us/azure/architecture/patterns/leader-election), you cannot synchronize if the code runs in different pods. As a result, you have to use an external method that is fraught with pitfalls during implementation[^1]. 
+
+Thankfully, Spring has done a lot of the hard work. All you need to do is provide it with a database connection and it will create a distributed lock. This example will show the lock with both Redis and JDBC.
+
+## Before You Begin
+
+Before you begin, you are going to need the following:
+
+- [Postgres](https://www.postgresql.org/) or [Redis](https://redis.io/)
+- A text editor or IDE of choice.
+  [JDK 16+](https://www.oracle.com/java/technologies/javase-downloads.html) or newer.(if you don't want to use the records that are setup in the example code Java 11+ will work)
+- An understanding of Spring Boot and dependency injection.
+
+Here is a [completed example on GitHub](https://github.com/estand64/distributed-lock) for you to view.
+
+## Package Imports
+
+To package imports, do the following:
+
+1. Import the necessary Spring Integration packages into your `pom.xml` file.
+2. Import the Spring Integration subpackages and drivers to connect to your datastore for either the Redis or JDBC version.
+3. All versions need the following:
+
```

Comment:
The number is off a little here, but then again my statements maybe should be rearranged to make it clearer because 1 and 3 are meant to be the same idea and then 2 and 4 are meant to be the same idea. I think removing step 2 and making step 1 a statement might be the way to go.

Reply:
I removed Step 2, then reordered the sequential numbering. I left Step 1 in because it needs to be done in order to do the new Step 2 (formerly Step 3).